### PR TITLE
✨ refactor: topbar and sidebar responsiveness

### PR DESCRIPTION
**Pull request by AntonyDerache**
# Pull request for Candy-doc components library

## Your checklist for this pull request

🚨 Please review the [guidelines for contributing](../../CONTRIBUTING.md) to
this repository.

- [X] Make sure you are requesting to **pull a feature/bugfix branch** (right
  side). Don't request your master!
- [X] Make sure you are making a pull request against the **main branch** (left
  side).
- [X] Check the commit's or even all commits' message styles matches our
  requested structure.
- [X] Check your code additions will fail neither code linting checks nor unit
  test.

## Description

Handle responsiveness of topbar and sidebar on mobile view

Tobpar:
- Add html content inside topbar to display a drawer available if viewport width is < 768px
- Add new property **isopen** to handle opening and closing of the drawer
- Add new slot **drawer-content** to add additional content in the topbar

Sidebar-doc:
- Fix some css values

❤️ Thank you!

### CODE REVIEW

File: src/components/Topbar/Topbar.ts [hunk 1]
```diff
@@ -32,23 +57,92 @@ export class CandyTopbar extends LitElement {
     }
   };
```
Comment (fvandenbulcke) [inline src/components/Topbar/Topbar.ts:58]:
```
defineIfDeviceIsMobile = () => {
      this.mobile = window.innerWidth < 768;
  };
```

File: src/components/Topbar/Topbar.ts [hunk 2]
```diff
@@ -32,23 +57,92 @@ export class CandyTopbar extends LitElement {
...
 
+  openDrawer() {
```
Comment (fvandenbulcke) [inline src/components/Topbar/Topbar.ts:60]: method `openMenu`  

ajouter this.isopen = true

File: src/components/Topbar/Topbar.ts [hunk 3]
```diff
@@ -32,23 +57,92 @@ export class CandyTopbar extends LitElement {
...
+    this.burgerButton.classList.add("active");
+    this.drawer.classList.remove("display-none");
+    this.drawer.classList.remove("topbar-drawer-close");
+    this.drawer.classList.add("topbar-drawer-active");
+  }
+
+  closeDrawer() {
```
Comment (fvandenbulcke) [inline src/components/Topbar/Topbar.ts:67]: method `closeMenu`

ajouter this.isopen = false

File: src/components/Topbar/Topbar.ts [hunk 4]
```diff
@@ -32,23 +57,92 @@ export class CandyTopbar extends LitElement {
...
+
+  handleBurgerClick = (openValue: boolean) => {
+    this.isopen = openValue;
+    this.emitValueChangeEvent(openValue);
+    if (openValue) {
+      this.openDrawer();
+    } else {
+      this.closeDrawer();
+    }
+  };
```
Comment (fvandenbulcke) [inline src/components/Topbar/Topbar.ts:93]:
```
handleMenuState = () => {
    if (this.isopen) {
      this.closeMenu();
    } else {
      this.openMenu();
    }
    this.emitValueChangeEvent(this.isopen);
  };
```

File: src/components/Topbar/Topbar.ts [hunk 5]
```diff
@@ -32,23 +57,92 @@ export class CandyTopbar extends LitElement {
...
-            ${this.mobile
-              ? html`<candy-logo-mark></candy-logo-mark>`
-              : html`<candy-logo-horizontal></candy-logo-horizontal>`}
-          </a>
+    const displaySlot = html`<slot></slot>`;
+
+    const displayBurger = html` <button
+      part="tobpar-burger"
+      id="topbar-burger"
+      @click=${() => this.handleBurgerClick(!this.isopen)}
```
Comment (fvandenbulcke) [inline src/components/Topbar/Topbar.ts:101]: @click=${this.handleMenuState}

**Review by fvandenbulcke — CHANGES_REQUESTED** [review]

File: src/components/Topbar/Topbar.ts (hunk 4 above)
Comment (AntonyDerache) [reply src/components/Topbar/Topbar.ts:93]: done

File: src/components/Topbar/Topbar.ts [hunk 6]
```diff
@@ -32,23 +57,92 @@ export class CandyTopbar extends LitElement {
...
+      <span></span>
+    </button>`;
+
+    const displayDrawer = html` <div
+      id="drawer"
+      part="tobpar-drawer"
+      class="topbar-drawer display-none"
+    >
+      <div
+        @click=${() => this.handleBurgerClick(!this.isopen)}
```
[the remark this answers is not in the document]
Comment (AntonyDerache) [reply src/components/Topbar/Topbar.ts:114]: same ,directly change isopen value to trigger update

Done

File: src/components/Topbar/Topbar.ts (hunk 5 above)
Comment (AntonyDerache) [reply src/components/Topbar/Topbar.ts:101]: directly change isopen value to trigger update,

Done

File: src/components/Topbar/Topbar.ts (hunk 3 above)
Comment (AntonyDerache) [reply src/components/Topbar/Topbar.ts:67]: Same than above

File: src/components/Topbar/Topbar.ts (hunk 1 above)
Comment (fvandenbulcke) [reply src/components/Topbar/Topbar.ts:54]: ok

File: src/components/Topbar/Topbar.ts (hunk 4 above)
Comment (fvandenbulcke) [reply src/components/Topbar/Topbar.ts:88]: ok

File: src/components/Topbar/Topbar.ts (hunk 2 above)
Comment (fvandenbulcke) [reply src/components/Topbar/Topbar.ts:56]: ok

File: src/components/Topbar/Topbar.ts (hunk 3 above)
Comment (fvandenbulcke) [reply src/components/Topbar/Topbar.ts:63]: ok

File: src/components/Topbar/Topbar.ts (hunk 5 above)
Comment (fvandenbulcke) [reply src/components/Topbar/Topbar.ts]: ok

File: src/components/Topbar/Topbar.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (fvandenbulcke) [reply src/components/Topbar/Topbar.ts]: ok

**Review by fvandenbulcke — APPROVED** [review]

**Review by Yugo-Cartron — APPROVED** [review]

**Pull request merged**